### PR TITLE
Feature/item dependency read only list for admin

### LINKED ISSUE #1064: Item dependency: read-only list for admin
## Overview
In the item (activity or skill) pages, add a tab in which the editor can configure the item dependencies.

## Acceptance criteria
- [ ] In item page, add a new tab "Dependencies" before "advanced parameters"
- [ ] This tab is only displayed to those who "can_edit >= 'all'"

This tab contains:
- [ ] A title "Prerequisites"
- [ ] An explanation: "Users will unlock access to this content if they solve **at least one** of the content below."
- [ ] A list of items (their title) from the ["prerequisites" service](https://franceioi-algorea.s3.eu-west-3.amazonaws.com/spec/v1.2.0.html#operation/itemPrerequisitesView) (no need for the `watched_group_id` parameter for now)
- [ ] only display those with `dependency_grant_content_view = true` 
- [ ] Each item has a pathless link, the path suggestion popup is shown on hover
- [ ] A remark, centered and with a smaller font (like this one:  Screenshot 2022-06-10 at 10 17 04) "Users will see the list of dependencies on the "content" tab only when they do not have access to the content yet. They will only see dependencies they are allowed to view already."

### PR DESCRIPTION
**Pull request by Iloveall**
## Description

Fixes #1064

## Notes (out of scope, known isues, hints for reviewing code, ...)  (optional)

...

## Test cases

- [x] Case 1:
  1. Given I am the usual user
  2. When I go to [this page](https://dev.algorea.org/branch/feature/item-dependency-read-only-list-for-admin/en/#/activities/by-id/1746497084338939783;path=4702,4102,686491270254002911,569694404892501336;parentAttempId=0/details/dependencies)
  3. Then I see list of dependencies
  4. And I hove on "Conversion de pouces en centimètres" item
  5. Then I tooltip with path suggestion

- [x] Case 2:
  1. Given I am the usual user
  2. When I go to [this page](https://dev.algorea.org/branch/feature/item-dependency-read-only-list-for-admin/en/#/activities/by-id/569694404892501336;path=4702,4102,686491270254002911;attempId=0/details/dependencies)
  3. Then I see the message: "There are no prerequisites for this activity. Users will have access to it only if they are given access manually or if the "default access" is configured as "open" in its parent chapter."

- [x] Case 3:
  1. Given I am the usual user
  2. When I go to [this page](https://dev.algorea.org/branch/feature/item-dependency-read-only-list-for-admin/en/#/skills/by-id/3000;path=;attempId=0/details/dependencies)
  3. Then I see the message: "You are not allowed to view this content"

### CODE REVIEW

File: src/app/modules/item/components/item-dependencies/item-dependencies.component.html [hunk 1]
```diff
@@ -0,0 +1,76 @@
+<ng-container *ngIf="itemData">
```
[the remark this answers is not in the document]
Comment (Iloveall) [reply src/app/modules/item/components/item-dependencies/item-dependencies.component.html:1]: Fixed

File: src/app/modules/item/components/item-dependencies/item-dependencies.component.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    }
+    this.showOverlaySubject$.next({ event, itemId, target: targetRef.nativeElement });
+  }
+
+  onMouseLeave(event: MouseEvent): void {
+    const target = event.target;
+    const relatedTarget = event.relatedTarget;
+    const keepOverlayOpened = target instanceof HTMLElement &&
+      relatedTarget instanceof HTMLElement &&
+      !!relatedTarget.closest('.alg-path-suggestion-overlay');
```
Comment (ThomasJuster) [inline src/app/modules/item/components/item-dependencies/item-dependencies.component.ts:67]: This code and in general the path overlay is very familiar, could we make a component of it?

**Review by ThomasJuster — COMMENTED** [review]
:+1:

File: src/app/modules/item/components/item-dependencies/item-dependencies.component.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  ngOnChanges(): void {
+    if (this.itemData) {
+      this.itemId$.next(this.itemData.item.id);
+    }
+  }
+
+  ngOnDestroy(): void {
+    this.itemId$.complete();
+    this.showOverlaySubject$.complete();
+    this.showOverlaySubscription.unsubscribe();
```
[the remark this answers is not in the document]
Comment (Iloveall) [reply src/app/modules/item/components/item-dependencies/item-dependencies.component.ts:53]: Fixed

**Review by ThomasJuster — COMMENTED** [review]
~See https://github.com/France-ioi/AlgoreaFrontend/pull/1097#discussion_r924597220~ :+1:

File: src/app/modules/item/components/item-dependencies/item-dependencies.component.ts (hunk 2 above)
Comment (Iloveall) [reply src/app/modules/item/components/item-dependencies/item-dependencies.component.ts]: https://github.com/France-ioi/AlgoreaFrontend/issues/1105

**Pull request merged**